src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/SignatureHelper.cs

```diff
@@ -12,6 +12,8 @@ internal static class MetadataSignatureHelper
     {
         internal static BlobBuilder GetLocalSignature(List<LocalBuilder> locals, ModuleBuilderImpl module)
         {
+            locals.Sort((l1, l2) => l1.LocalIndex - l2.LocalIndex); // sort by created order
```

Comment:
Does it make sense to add a `List<LocalBuilder>` field to `ILGenerator` and add to it whenever `DeclareLocal()` is called and then use that? That should prevent the need for sorting, but requires some dual tracking.

Reply:
That is an another way, if you prefer that let me know. For me there is not much difference. Note that this `GetLocalSignature(...)` method only called once per method.